bpf/bpf_lxc.c

```diff
@@ -194,6 +143,21 @@ static __always_inline int handle_ipv6_from_lxc(struct __ctx_buff *ctx, __u32 *d
 
 	reason = ret;
 
+#ifdef ENABLE_PER_PACKET_LB
+	{
+		/* Restore data from service handling. */
+		__u32 meta = ctx_load_meta(ctx, CB_CT_STATE);
+
+		ct_state_new.rev_nat_index = meta >> 16;
+		ct_state_new.loopback = meta & 1;
+		hairpin_flow = ct_state_new.loopback;
```

Comment:
The logic to encode/decode the values from slot `CB_CT_STATE` could be moved to helper functions. Especially given we have the exact same logic on the IPv4 side.

Reply:
Done!